src/transport/SecureSessionMgr.cpp

```diff
@@ -119,9 +123,31 @@ Transport::Type SecureSessionMgr::GetTransportType(NodeId peerNodeId)
 CHIP_ERROR SecureSessionMgr::SendMessage(SecureSessionHandle session, PayloadHeader & payloadHeader,
                                          System::PacketBufferHandle && msgBuf, EncryptedPacketBufferHandle * bufferRetainSlot)
 {
+    PeerConnectionState * state = nullptr;
     PacketHeader unusedPacketHeader;
-    return SendMessage(session, payloadHeader, unusedPacketHeader, std::move(msgBuf), bufferRetainSlot,
-                       EncryptionState::kPayloadIsUnencrypted);
+
+    state = GetPeerConnectionState(session);
+    VerifyOrReturnError(state != nullptr, CHIP_ERROR_NOT_CONNECTED);
+
+    if (!IsControlMessage(payloadHeader) && !state->GetSessionMessageCounter().GetPeerMessageCounter().IsSynchronized())
```

Comment:
Doesn't message counter sync only kick-in on receipt of a group-key encrypted message? This code-path here seems to indicate that it is initiated on transmission, which isn't the case as per the current spec.

Reply:
For other situation, `IsSynchronized` will return true, so it won't go into this branch.

Reply:
But it should _never_ need to go into this branch for any transmission, group-key or not.